### PR TITLE
Add specification for circular arc segments (in degrees)

### LINKED ISSUE #11: Arcs / curves
Should we support arcs? Does LFF support them?

If yes, what types of arcs? Only circular arcs?

Should we recommend or require that implementations provide a way to flatten arcs, returning them as polylines?

Would Bézier curves (quadratic / cubic) be totally out of scope? Efficient flattening is not *that* hard. [Here](https://github.com/nical/lyon/blob/master/bezier/src/flatten_cubic.rs) is the implementation by [Lyon](https://docs.rs/lyon_bezier/0.8.0/lyon_bezier/) for example, based on [this](http://cis.usouthal.edu/~hain/general/Publications/Bezier/Bezier%20Offset%20Curves.pdf) paper. (Ignore the Math, the implementation should be easier to follow.)

### PR DESCRIPTION
**Pull request by ubruhin**
An alternative to #18.

Fixes #11.

### CODE REVIEW

File: SPECIFICATION.md [hunk 1]
```diff
@@ -113,6 +113,21 @@ LETTER A (U+0041)":
     0.8333,2.5;5.1666,2.5
     0,0;3,9;6,0
 
+In addition to straight lines, polylines can contain circular arc segments.
+This is done by adding a third parameter, called *bulge*, to the coordinate
+pair of the end coordinate of an arc segment. It specifies the included angle of
+the circular arc segment. The bulge is a value from -9 to +9 and represents an
+angle in degrees from -180° to +180°. A positive angle describes a counter
+clockwise arc segment from the start point to the end point, while a negative
```
Comment (dbrgn) [inline SPECIFICATION.md]: Which point is the start point and the end point? If a coordinate pair is followed by a bulge, is the circle segment from the previous point to the current point, or from the current point to the next point?

File: SPECIFICATION.md [hunk 2]
```diff
@@ -113,6 +113,21 @@ LETTER A (U+0041)":
     0.8333,2.5;5.1666,2.5
     0,0;3,9;6,0
 
+In addition to straight lines, polylines can contain circular arc segments.
+This is done by adding a third parameter, called *bulge*, to the coordinate
+pair of the end coordinate of an arc segment. It specifies the included angle of
+the circular arc segment. The bulge is a value from -9 to +9 and represents an
+angle in degrees from -180° to +180°. A positive angle describes a counter
```
Comment (dbrgn) [inline SPECIFICATION.md]: Can you add a note that the angle specifies the central angle?

**Review by dbrgn — CHANGES_REQUESTED** [review]
I like this approach better than the other one. It needs an illustration though.

I think @rnestler is currently working on one.

**Comment by ubruhin** [comment]
> I think @rnestler is currently working on one.

@rnestler are you? :)

**Comment by dbrgn** [comment]
I am :)

**Comment by dbrgn** [comment]
This is now implemented here: https://fontobene.dbrgn.ch/pathvis/

@ubruhin is this behavior how you meant it?

**Comment by ubruhin** [comment]
> @ubruhin is this behavior how you meant it?

Yes, exactly that way :smiley:  Except that an angle of 0 should also work :wink:

**Comment by dbrgn** [comment]
:D Fixed.

Can you add the clarifications as per the comments above? If you want, you can use the following examples:

`8,8;1,2,4` (central angle is 80°)

img

`8,8;1,2,6` (central angle is 120°)

img

`8,8;1,2,-6` (central angle is -120°)

img

`8,8;1,2,-9` (central angle is -180°)

img

And here a simple illustration of the angle (based on Wikipedia):

segment

(SVG file: https://tmp.dbrgn.ch/circular_segment.svg)

File: SPECIFICATION.md (hunk 1 above)
Comment (ubruhin) [reply SPECIFICATION.md]: Should this be clear with this sentence of the explanation?

> This is done by adding a third parameter, called *bulge*, to the coordinate pair of the **end coordinate** of an arc segment.

**Comment by ubruhin** [comment]
Better? [rendered](https://github.com/fontobene/fontobene/blob/50604a0a283b92af16c2cbf082dfdbe264cb176f/SPECIFICATION.md#circular-arc-segments)

**Comment by ubruhin** [comment]
Ping @dbrgn

**Review by dbrgn — APPROVED** [review]

**Pull request merged**